smartapps/initialstate-events/initial-state-event-streamer.src/initial-state-event-streamer.groovy

```diff
@@ -331,58 +330,57 @@ def genericHandler(evt) {
 // after a specified amount of time to reduce the load on ST servers
 def flushBuffer() {
 	log.trace "About to flush the buffer on schedule"
-	if (atomicState.eventBuffer != null && atomicState.eventBuffer.size() > 0) {
+	if (state.eventBuffer != null && state.eventBuffer.size() > 0) {
 		tryShipEvents()
 	}
 }
 
 def eventHandler(name, value) {
-	log.debug atomicState.eventBuffer
-
 	def eventBuffer = atomicState.eventBuffer
-	def epoch = now() / 1000
-	
-	// if for some reason this code block is being run
-	// but the SmartApp wasn't propery setup during install
-	// we need to set initialize the eventBuffer.
-	if (!atomicState.eventBuffer) {
-		atomicState.eventBuffer = []
+	if (!eventBuffer) {
+		// Initialize the eventBuffer since the SmartApp wasn't setup properly
+		eventBuffer = []
 	}
 	eventBuffer << [key: "$name", value: "$value", epoch: "$epoch"]
-	
-	log.debug eventBuffer
-
 	atomicState.eventBuffer = eventBuffer
 
-	if (eventBuffer.size() >= 10) {
+	log.debug eventBuffer
+
+	// Try to post to Initial State every 10 events, doing mod in case there are failures and we didn't flush
+	if (eventBuffer.size() % 10 == 0) {
 		tryShipEvents()
 	}
 }
 
 // a helper function for shipping the atomicState.eventBuffer to Initial State
 def tryShipEvents() {
-
 	// can't ship events if there is no grokerSubdomain
-	if (atomicState.grokerSubdomain == null || atomicState.grokerSubdomain == "") {
+	if (state.grokerSubdomain == null || state.grokerSubdomain == "") {
 		log.error "streaming url is currently null"
+		// Avoid the eventBuffer from growing unbounded if streaming url isn't set
+		atomicState.eventBuffer = []
 		return
 	}
 	// can't ship if access key and bucket key are null, so finish trying
-	if (atomicState.accessKey == null || atomicState.bucketKey == null) {
+	if (state.accessKey == null || state.bucketKey == null) {
+		// Avoid the eventBuffer from growing unbounded if no accessKey or bucketKey is set
+		atomicState.eventBuffer = []
 		return
 	}
 
+	def eventBuffer = atomicState.eventBuffer
 	def eventPost = [
-		uri: "https://${atomicState.grokerSubdomain}.initialstate.com/api/events",
+		uri: "https://${state.grokerSubdomain}.initialstate.com/api/events",
 		headers: [
 			"Content-Type": "application/json",
-			"X-IS-BucketKey": "${atomicState.bucketKey}",
-			"X-IS-AccessKey": "${atomicState.accessKey}",
+			"X-IS-BucketKey": "${state.bucketKey}",
+			"X-IS-AccessKey": "${state.accessKey}",
```

Comment:
If the `bucketKey` and `accessKey` are going to be changed to be on `state` as opposed to `atomicState` they need to be changed also in the `setAccessKey`, `getAccessKey`, `setBucketKey`, and `getBucketKey` methods as well.